src/components/Testimonials/index.jsx

```diff
@@ -30,6 +31,29 @@ const tweets = [
 ];
 
 export function Testimonials() {
+  const [isMobile, setIsMobile] = useState(false);
+  const [visibleTweets, setVisibleTweets] = useState(8);
+  const [showAll, setShowAll] = useState(false); // State to toggle between showing all and limited tweets
+
+  useEffect(() => {
+    const handleResize = () => {
+      setIsMobile(window.innerWidth < 768);
+    };
+
+    handleResize(); // Check initially
+    window.addEventListener('resize', handleResize);
+
+    return () => window.removeEventListener('resize', handleResize);
+  }, []);
+
+  useEffect(() => {
+    setVisibleTweets(isMobile ? (showAll ? tweets.length : 8) : tweets.length);
+  }, [isMobile, showAll]);
```

Comment:
Why do you need this to be a state? I think you are over complicating this, can you think of a simpler way of doing so without having nested conditionals and a state instead?

Reply:
Got it